### PR TITLE
F/tests

### PR DESCRIPTION
**Pull request by jwilles**
This pull request introduces a comprehensive testing infrastructure for the VBLL library, including automated CI setup, configuration files, fixtures, and documentation. The changes enable robust, reproducible, and flexible testing using pytest, and provide clear guidance for contributors on running and extending the test suite.

**Testing Infrastructure and Automation**
- Added a GitHub Actions CI workflow (`.github/workflows/ci.yml`) to automatically run tests on pushes and pull requests to `main`/`master` branches. The workflow sets up Python, installs dependencies (including PyTorch), and runs the test suite with pytest.
- Introduced a test runner script (`run_tests.py`) that provides a command-line interface for running the test suite with options for quick/unit/integration/benchmark tests, parallel execution, coverage, and optional GPU/JAX support.

**Test Suite Configuration and Fixtures**
- Added `pytest.ini` to configure pytest: specifies test discovery patterns, verbosity, warning filters, and custom markers for slow, GPU, integration, and unit tests.
- Created `tests/conftest.py` with a comprehensive set of fixtures for devices, seeds, batch sizes, sample data, and common parameter sets for classification and regression layers, supporting reproducible and parameterized tests.

**Documentation and Developer Guidance**
- Added `tests/README.md` detailing test structure, categories, running instructions, coverage, benchmarks, and troubleshooting tips, guiding contributors on best practices for testing and extending the suite.
- Updated the main `README.md` with instructions for installing test dependencies and running tests using pytest.

**Project Configuration**
- Updated `pyproject.toml` to include `pytest` as a development dependency, ensuring test requirements are managed via Poetry.
- Added `tests/__init__.py` to mark the tests directory as a package.

These changes collectively establish a solid foundation for automated and manual testing, improving code quality and maintainability.

### CODE REVIEW

**Review by jmesharrison — APPROVED** [review]

File: tests/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        for epoch in range(5):
+            optimizer.zero_grad()
+            
+            result = layer(data['x'])
+            train_loss = result.train_loss_fn(data['y_classification'])
+            
+            train_loss.backward()
+            optimizer.step()
+            
+            # Check that loss is finite and decreasing
```
Comment (MJordahn) [inline tests/test_integration.py:37]: No test for decreasing here currently

Comment (MJordahn) [reply tests/test_integration.py:37]: This might fail for large learning rates though (decreasing check). Suggest either reducing learning rate or simply not checking for decreasing.

File: tests/test_regression.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+        pred_dist = layer.predictive(x)
+        
+        # Test sampling
+        samples = pred_dist.sample((10,))
+        assert samples.shape == (10, data['batch_size'], data['output_features'])
+        
+        # Test log probability
+        y = data['y_regression']
+        log_prob = pred_dist.log_prob(y)
+        assert log_prob.shape == (data['batch_size'], data['output_features'])
```
Comment (MJordahn) [inline tests/test_regression.py:117]: Should shape of log_prob here be (data['batch_size'], data['output_features']) or will it always just be (data['batch_size'], 1)?
